### DISCUSSION TITLE
How to know all async calls are complete when using async thunk?

### DISCUSSION THREAD

**Discussion by heart2friend** [reduxjs/redux-toolkit#2902] [Q&A]
Hi All, I have to keep an option selected by default in dropdown based on some conditions and it will sort data on first load based on selected option. sorting can happen only when all child requests are complete. How we can determine whether all async call made via async thunk are complete and data is loaded? so I can trigger sort.

**Here** is an example

I have sales order, each sales order can have multiple product. each product can have multiple bin. once all bins are loaded I want to trigger sort. I get products with sales order then it's loaded in component. on Product component giving call to get bin details of product. bins will be rendered in bin component. 

Really appreciate the help.

**Comment by markerikson** [comment]
Generally, `await dispatch(someAsyncThunk())`:

- https://redux.js.org/tutorials/essentials/part-5-async-logic#checking-thunk-results-in-components
- https://redux.js.org/tutorials/fundamentals/part-7-standard-patterns#thunks-and-promises

**Comment by heart2friend** [comment]
Hi @markerikson  thank you for your prompt response. I wasn't aware about this. certainly it help me solve some other scenarios :). 

I am able to detect whether all async calls are complete or not.  I have used "react-promise-tracker" to track "promiseInProgress" on parent component. I have wrapped dispatch inside trackPromise method. 

![image](https://user-images.githubusercontent.com/33868668/201598195-c26a1519-3290-404a-a205-0d25902238d3.png)
